src/CarbonAware.WebApi/src/Controllers/CarbonAwareController.cs

```diff
@@ -159,4 +159,14 @@ private async Task<IActionResult> GetEmissionsDataAsync(Dictionary<string, objec
         var response = await _aggregator.GetEmissionsDataAsync(props);
         return response.Any() ? Ok(response) : NoContent();
     }
+
+    private IEnumerable<Location> CreateLocationsFromQueryString(string[] queryStringLocations)
+    {
+        var locations = queryStringLocations
```

Comment:
A simple check `quesryStringLocation == null || queryStringLocations.Length == 0` should return the exception.

Reply:
I wish this were true 🙃  dotnet will convert `my/api/path?location=` to an array with 1 element: `[null]`

`queryStringLocations == null` returns false
`queryStringLocations.Length` returns 1